TATB - 39 [BUG]: Wrap Dialog Data into one config object for more readability

PrabhatJTDeveloper opened the pull request.

PrabhatJTDeveloper committed:
- TATB 39
    - for wraping dialog data into one config object changed the modal component ts file.
    - in modal component html file use that object .
    - and the modal is opening in signout which is in header component and skip button which is in quiz component so changed this two components.
    - add one object signoutModal in dialogData.json file.

PrabhatJTDeveloper committed:
- TATB 39
    - solved the conflicts
    - removed unused class and id from the button tag of header component html file.
    - added about Modal data to the dialogeData.json

PrabhatJTDeveloper committed:
- TATB 39
     - removed ?.length from the login component ts file.
     - removed import of dilogData.json from dialog service
     - updated modal component html file.

PrabhatJTDeveloper committed:
- TATB 39
  - change the mapJsonData() to getQuestionData() function name in quiz component ts file and in statquiz component ts file mapJsonData() to getInstruction().

charmijtdev commented on src/app/authorization/login/login.component.ts:
```diff
@@ -45,9 +45,8 @@ export class LoginComponent implements OnInit, OnDestroy {
-        localStorage.getItem('registerUser') as string
-      );
+      let data: any = localStorage.getItem('registerUser');
+      this.userData = JSON.parse(data);
```
could you please change this to JSON.parse(localstore.getitem('register user')as string)

PrabhatJTDeveloper replied:
yes, i changed it.

charmijtdev commented on src/app/component/header/header.component.html:
```diff
@@ -1,22 +1,62 @@
+      href="#"
+      class="navbar-brand h1"
+      routerLink="/dashboard"
+      style="margin: 0px"
```
why inline css

PrabhatJTDeveloper replied:
yes, i removed it

charmijtdev commented on src/app/component/header/header.component.html:
```diff
@@ -1,22 +1,62 @@
+>
+  <div class="container-fluid">
+    <a
+      href="#"
```
no need for href tag

PrabhatJTDeveloper replied:
ok, i removed it.

charmijtdev commented on src/app/component/header/header.component.html:
```diff
@@ -1,22 +1,62 @@
+      <img
+        class="d-inline-block"
+        src="assets/img/logo1.png"
+        width="35"
```
why width height on same please change to external css

PrabhatJTDeveloper replied:
yes. i removed it.

charmijtdev commented on src/app/component/header/header.component.html:
```diff
@@ -1,22 +1,62 @@
+    <div class="navigation">
+      <div class="container-fluid">
+        <div class="collapse navbar-collapse">
+          <img src="assets/img/logo1.png" alt="" class="img-responsive" />
```
please provide alt here

PrabhatJTDeveloper replied:
ok, i changed it.

charmijtdev commented on src/app/component/header/header.component.html:
```diff
@@ -1,22 +1,62 @@
+              type="button"
+              class="nav-link dropdown-toggle"
+              id="dropdownBasic1"
+              style="color: rgb(116, 116, 116)"
```
why inline css please change this

PrabhatJTDeveloper replied:
ok, i changed it.

charmijtdev commented on src/app/component/header/header.component.html:
```diff
@@ -1,22 +1,62 @@
+            <div
+              ngbDropdownMenu
+              aria-labelledby="dropdownBasic1"
+              style="min-width: auto; border: none"
```
why inline css ?

PrabhatJTDeveloper replied:
i removed it.

charmijtdev commented on src/app/component/header/header.component.ts:
```diff
@@ -27,12 +31,31 @@ export class HeaderComponent {
     let userId: any = this.authenticationService.getUser();
     this.userName = this.userData?.find(
-      (data: any) => data?.id == userId
+      (data: any) => data?.id
```
why removed condtiion ? how i can get the specified user ?

PrabhatJTDeveloper replied:
yes, i added it.

charmijtdev commented on src/app/component/header/header.component.ts:
```diff
@@ -27,12 +31,31 @@ export class HeaderComponent {
+      this.dialogService
+        .openDialog(configData)
+        .then((value) => {
+          if (value) {
```
wht is use for value here ?

PrabhatJTDeveloper replied:
there is no use for it so i removed it.

charmijtdev commented on src/app/dialog-service/dialog.service.ts:
```diff
@@ -1,39 +1,30 @@
-  dialogData = { ...dialogData };
+  constructor(
+    private modalService:NgbModal,
+  
```
please remove this empty space and remove comma after NgbModal

PrabhatJTDeveloper replied:
yes, i removed it.

charmijtdev commented on src/app/dialog-service/modal/modal/modal.component.html:
```diff
@@ -1,4 +1,64 @@
+</div>
+</div>
+
+<!-- <div class="container-fluid">
```
please remove this commented code

PrabhatJTDeveloper replied:
yes, i removed it.

charmijtdev commented on src/app/dialog-service/modal/modal/modal.component.html:
```diff
@@ -1,4 +1,64 @@
+  >
+    {{ configData?.yes }}
+  </button>
+</div>
```
do code formatting

PrabhatJTDeveloper replied:
yes, i have done this.

PrabhatJTDeveloper committed:
- TATB 39
    - removed inline css and give the extenal css
    - added the userId in the login component ts file
    - removed unnecassary line space and comma from the dialog service ts file
    - formated the code in modal component html file and also removed commented code.

charmijtdev commented on src/app/authorization/login/login.component.ts:
```diff
@@ -6,24 +6,21 @@ import { AuthenticationService } from 'src/app/service/authentication.service';
   selector: 'app-login',
   templateUrl: './login.component.html',
-  styleUrls: ['./login.component.scss']
+  styleUrls: ['./login.component.scss'],
```
please remove comma

charmijtdev commented on src/app/component/header/header.component.html:
```diff
@@ -1,22 +1,41 @@
-          </a>
+<nav
+  class="navbar fixed-top navbar-expand-sm navbar-light bg-light"
+  style="padding: 0px"
```
please remove inline styling

charmijtdev commented on src/app/component/header/header.component.ts:
```diff
@@ -1,25 +1,27 @@
   selector: 'app-header',
   templateUrl: './header.component.html',
-  styleUrls: ['./header.component.scss']
+  styleUrls: ['./header.component.scss'],
```
please remove comma

charmijtdev commented on src/app/component/header/header.component.ts:
```diff
@@ -30,9 +32,20 @@ export class HeaderComponent {
     )?.fullName;
   }
+
+  about() {
```
change function name to openAboutDialog()

PrabhatJTDeveloper committed:
- TATB 39
    - Removed extra comma from the login and header component ts file and also change the function name for opening dialog modal in header component ts file.
    - change the class name in header component scss file.
    - make change in dialogData.json file
